### PR TITLE
Add support for the latest bleach version

### PR DESCRIPTION
**Pull request by oscarmcm**
This is a continuation of #1253, an improved version of #1255, and fixes:

- #1259 
- #1257 
- #1251

### CODE REVIEW

**Review by benjaoming — APPROVED** [review]
These look like good changes. Wondering if they work with some old version of Bleach.

**Comment by oscarmcm** [comment]
> These look like good changes. Wondering if they work with some old version of Bleach.

I dunno, This is a very breaking change... unless we add middleware to transform old Django wiki settings (list) into a frozenset; but from what I know this wouldn't work out of the box.

**Comment by benjaoming** [comment]
Maybe we should push the lower bound on bleach. Our tests suggest that it's not a problem wrt. Django and Python versions.

The only way that it can cause problems is with other dependencies for users of django-wiki. But I think the chances are narrow, and those might as well consider upgrading Bleach.

File: pyproject.toml [hunk 1]
```diff
@@ -40,7 +40,7 @@ dependencies = [
     # Django version support is added one minor release at a time
     "Django>=2.1,<4.2",
     # bleach has been pretty stable, hence the loose pin
-    "bleach[css]>=5,<6",
+    "bleach[css]>=5,<7",
```
Comment (benjaoming) [inline pyproject.toml:43]:
```suggestion
    "bleach[css]>=6,<7",
```

File: pyproject.toml [hunk 2]
```diff
@@ -194,7 +194,7 @@ pull = [
 [tool.hatch.envs.docs]
 dependencies = [
     "Django>=3.1.13",
-    "bleach>=3.3.0,<5.1",
+    "bleach>=3.3.0,<7",
```
Comment (benjaoming) [inline pyproject.toml:197]:
```suggestion
    "bleach>=6,<7",
```

**Comment by benjaoming** [comment]
Hmm, the demo currently looks like this. And we haven't changed anything to Bootstrap. I'm wondering if this is the new Bleach version removing stuff that is intended to be included:

image

However, in the case of "locked", that text isn't supposed to be visible at all.

**Comment by oscarmcm** [comment]
@benjaoming related to the image, I just changed the image tag to use an external source instead of the internal, and thus there's no way for that image to align with plain markdown.

**Comment by benjaoming** [comment]
@oscarmcm roger that, hmm... it looks kind of bad for the demo.. but unrelated to this PR... maybe we can get an image for the fixtures somehow, so it'll look better. Perhaps a management command to setup the fixtures? I've had good experiences in other projects of creating a management command for fixtures to deal with complex stuff.

**Comment by miguelgr** [comment]
Hi @oscarmcm , will these changes be available in a patch version for v9.x ? or will be included in [0.10](https://pypi.org/project/wiki/0.10b1/) 
If I am not wrong, currently by `pip installing wiki` the default version of wiki is v9 which doesn't handle the TypeError for the bleach.ALLOWED_TAGS.
Thanks in advance.

**Comment by benjaoming** [comment]
@miguelgr I think we can release this in a 0.10b2 - would you mind giving some feedback on it if we put up that release?

**Comment by miguelgr** [comment]
I did try `pip install --pre wiki` and I got the site up and running even with the fixes for 3.11 included in #1232, also I guess for the 0.10 release.

**Comment by benjaoming** [comment]
@miguelgr the 0.10b1 version didn't contain this PR AFAICT.

**Comment by miguelgr** [comment]
It did fix the issue that 0.9 has with bleach 6.
Where `wikipedia` is a just created django project with the wiki installed.

image

**Pull request merged**